### PR TITLE
[Feature-6071][common|dao|datasource-plugin] implement the dynamic loading of data source plug-ins (#6071)

### PR DESCRIPTION
**Pull request by Narcasserun**
- The data source plug-in loads dynamically through the URL and unload dynamically
- Unified data source connection management
- add datasource plug-in client
- add data source plug-in package

### CODE REVIEW

**Review by CalvinKirs — COMMENTED** [review]
Some jars are no longer needed, you need to delete related declarations

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-postgresql/pom.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    <parent>
+        <artifactId>dolphinscheduler-datasource-plugin</artifactId>
+        <groupId>org.apache.dolphinscheduler</groupId>
+        <version>1.3.6-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>dolphinscheduler-datasource-postgresql</artifactId>
+    <name>${project.artifactId}</name>
+
```
Comment (CalvinKirs) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-postgresql/pom.xml:30]: The plugin packaging method needs to be changed to <packaging>dolphinscheduler-plugin</packaging>

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-postgresql/pom.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    <dependencies>
+
+        <dependency>
+            <groupId>org.apache.dolphinscheduler</groupId>
+            <artifactId>dolphinscheduler-dao</artifactId>
+            <scope>provided</scope>
+        </dependency>
```
Comment (CalvinKirs) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-postgresql/pom.xml:37]: Plugins should remain independent and not depend on the kernel module

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/provider/DataSourceClientProvider.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+public class DataSourceClientProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(DataSourceClientProvider.class);
+
+    private static final JdbcDriverManager jdbcDriverManagerInstance = JdbcDriverManager.getInstance();
+
+    public static DataSourceClient createDataSourceClient(BaseConnectionParam baseConnectionParam, DbType dbType) {
+        logger.info("Creating the createDataSourceClient. JdbcUrl: {} ", baseConnectionParam.getJdbcUrl());
+        //Check jdbc driver location
+        checkDriverLocation(baseConnectionParam, dbType);
```
Comment (CalvinKirs) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/provider/DataSourceClientProvider.java:53]: It is better to use the spi module's plug-in loading method uniformly

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-postgresql/pom.xml (hunk 1 above)
Comment (Narcasserun) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-postgresql/pom.xml:30]: Yes, I'll modify it

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-postgresql/pom.xml (hunk 2 above)
Comment (Narcasserun) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-postgresql/pom.xml:37]: Basically, those who rely on datasource plugin will rely on Dao. At present, there is no major problem in terms of reusability and scalability. If there are more complex scenarios in the future, they can be split

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/provider/DataSourceClientProvider.java (hunk 3 above)
Comment (Narcasserun) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/provider/DataSourceClientProvider.java:53]: Thank you for your suggestion. I will try to use the SPI module's plug-in loading method uniformly

**Pull request closed without merging**